applications/OptimizationApplication/python_scripts/controls/control.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+
+class Control(ABC):
+    """Base abstract control class.
+
+    The base abstract control class has the following responsibilities.
+        1. Initalizes and finalizes the control.
+        2. Mapping physical space gradients to control space gradients.
```

Comment:
I would make it more elaborated by saying  "Mapping consistently physical space gradients if a mapping/filtering/parametrization is involved, otherwise pass "

Reply:
consistently is also not correct. One can filter only gradients and not updates if he wants. 

```suggestion
        2. Maps physical space gradients to control space gradients, if needed. Otherwise, it passes. 
```